### PR TITLE
General Accuracy Improvements

### PR DESCRIPTION
**Pull request by The1UPGuy**
I've made a good amount of tweaks to make TPP a bit more accurate and match footage more. The exact changes can be found on TPPCHANGELIST.txt in the main folder, that .txt file and some commented out credits to others that helped with code should probably be removed if these changes are added to the main repo. If any questions emerge about the changes i'll be sure to answer them.

### CODE REVIEW

**Comment by The1UPGuy** [comment]
Whoops realized I made a mistake in the change list, "Mario's movement speed is directly based on movement speed" should be "Mario's movement speed is directly based on how far the stick is held", my bad.

**Comment by Fluvian** [comment]
Any reason the trees were changed?

**Comment by The1UPGuy** [comment]
I've changed them because when looking at footage from multiple builds the trees always seem pretty dark in them, that darker color can be replicated by fully importing the trees from the gigaleak to SM64.
trees

**Comment by Fluvian** [comment]
The trees are actually darker in the builds, the file you imported seems to be based off of those trees, the trees in SW were lighter, there doesn't seem to be any difference in lighting either so it's just the texture being different, which is wrong. The texture in TPP right now is the correct SW95 one.

**Comment by The1UPGuy** [comment]
Oh I see. I'll revert them back to how they were in TPP.

File: src/game/mario_actions_moving.c [hunk 1]
```diff
@@ -549,26 +549,28 @@ void anim_and_audio_for_walk(struct MarioState *m) {
...
                         m->actionTimer = 2;
+                        m->pretiptimer = 0;
                     } else {
-                        val14 = (s32)(val04 * 0.75f * 0x10000);
+                        val14 = (m->intendedMag * 10000.0f) * 2.5f;
                         set_mario_anim_with_accel(m, MARIO_ANIM_START_TIPTOE, val14);
                         play_step_sound(m, 7, 22);
-                        if (is_anim_past_frame(m, 23)) {
-                            m->actionTimer = 2;
+                        m->pretiptimer++;
```
Comment (Fluvian) [inline src/game/mario_actions_moving.c:559]: Rename `pretiptimer` to `pretipTimer` to match the other struct variable names.

File: src/game/mario_actions_moving.c [hunk 2]
```diff
@@ -473,6 +473,10 @@ void update_walking_speed(struct MarioState *m) {
         m->forwardVel = 48.0f;
     }
 
+    if (m->forwardVel >= targetSpeed) { //code by Guestpiki
```
Comment (Fluvian) [inline src/game/mario_actions_moving.c:476]: Fix the indentation here, use spaces and not tabs.

Also, could you remove the comment credits? I'll add them to the README.

File: src/game/mario_actions_cutscene.c [hunk 3]
```diff
@@ -1050,13 +1050,11 @@ s32 act_exit_land_save_dialog(struct MarioState *m) {
 }
 
 s32 act_death_exit(struct MarioState *m) {
+    if (m->actionTimer == 15) {
+    play_sound(SOUND_MARIO_HOOHOO, m->marioObj->header.gfx.cameraToObject);
```
Comment (Fluvian) [inline src/game/mario_actions_cutscene.c:1054]: Fix the indentation here.

File: src/game/mario_actions_airborne.c [hunk 4]
```diff
@@ -1361,9 +1366,14 @@ s32 act_forward_rollout(struct MarioState *m) {
 
 s32 act_backward_rollout(struct MarioState *m) {
     if (m->actionState == 0) {
-        m->vel[1] = 15.0f;
-        m->actionState = 1;
-    }
+	    if (m->intendedMag > 0) {
```
Comment (Fluvian) [inline src/game/mario_actions_airborne.c:1369]: Fix the indentations here.

File: src/game/mario_actions_airborne.c [hunk 5]
```diff
@@ -1320,9 +1320,14 @@ s32 act_air_hit_wall(struct MarioState *m) {
 
 s32 act_forward_rollout(struct MarioState *m) {
     if (m->actionState == 0) {
-        m->vel[1] = 15.0f;
-        m->actionState = 1;
-    }
+	    if (m->intendedMag > 0) {
```
Comment (Fluvian) [inline src/game/mario_actions_airborne.c:1323]: Fix the indentations here too.

File: src/game/mario.c [hunk 6]
```diff
@@ -432,23 +432,23 @@ s32 mario_get_floor_class(struct MarioState *m) {
 }
 
 // clang-format off
-s8 sTerrainSounds[7][6] = {
+s8 sTerrainSounds[7][6] = { //code by Crustian
```
Comment (Fluvian) [inline src/game/mario.c:435]: Fix the indentation here so the "S" of the `SOUND_TERRAIN_X` matches those below it.

Also I don't mind giving credit but it's just a few value changes I would have done anyway if we found the footstep sounds.

File: src/game/interaction.c [hunk 7]
```diff
@@ -1188,16 +1187,21 @@ u32 interact_bully(struct MarioState *m, UNUSED u32 interactType, struct Object
...
-
-        push_mario_out_of_object(m, o, 5.0f);
-        drop_and_set_mario_action(m, bully_knock_back_mario(m), 0);
 #ifdef VERSION_SH
         queue_rumble_data(5, 80);
 #endif
+        push_mario_out_of_object(m, o, 5.0f);
+
+        drop_and_set_mario_action(m, bully_knock_back_mario(m), 0);
+        if (m->forwardVel < 0.0f) {
```
Comment (Fluvian) [inline src/game/interaction.c:1196]: Indentation again.

File: src/game/interaction.c [hunk 8]
```diff
@@ -864,10 +861,12 @@ u32 interact_warp(struct MarioState *m, UNUSED u32 interactType, struct Object *
                 queue_rumble_data(12, 80);
             }
 #else
+        if (gCurrLevelNum != LEVEL_CASTLE) {
```
Comment (Fluvian) [inline src/game/interaction.c:864]: Indentation here, move the if to the same the same column as the closing bracket above the `#else` directive.

File: src/game/interaction.c [hunk 9]
```diff
@@ -485,10 +482,10 @@ u32 bully_knock_back_mario(struct MarioState *mario) {
     mario->pos[0] = marioData.posX;
     mario->pos[2] = marioData.posZ;
 
-    bully->oMoveAngleYaw = newBullyYaw;
+    /*bully->oMoveAngleYaw = newBullyYaw;
```
Comment (Fluvian) [inline src/game/interaction.c:485]: You can just remove this entirely.

File: src/game/behaviors/bowser_bomb.inc.c [hunk 10]
```diff
@@ -35,16 +35,18 @@ void bhv_bowser_bomb_explosion_loop(void) {
...
-        o->oOpacity = 0;
-
     o->oPosY += o->oVelY;
 
+    if (o->oTimer < 25)
+    cur_obj_scale((f32) o->oTimer / 14.0f * 9.0 + 1.0);
+
     if (o->oTimer == 28)
+       o->oPosY = o->oPosY;
+       o->oVelY = 0.0f;
```
Comment (Fluvian) [inline src/game/behaviors/bowser_bomb.inc.c:48]: **BUG!**

This line will be run regardless of the if statement, if statements without brackets will only be valid for one line after it, add brackets to the if statement here. 

If the bomb smoke behaves not as intended after this change then remove the brackets again but add a line between the two and move the `o->oVelY = 0.0f;` statement 4 spaces to the left.

File: src/game/behaviors/bowser_bomb.inc.c [hunk 11]
```diff
@@ -35,16 +35,18 @@ void bhv_bowser_bomb_explosion_loop(void) {
...
         o->oAnimState++;
 
-    o->oOpacity -= 10;
-    if (o->oOpacity < 10)
-        o->oOpacity = 0;
-
     o->oPosY += o->oVelY;
 
+    if (o->oTimer < 25)
+    cur_obj_scale((f32) o->oTimer / 14.0f * 9.0 + 1.0);
```
Comment (Fluvian) [inline src/game/behaviors/bowser_bomb.inc.c:44]: Indentation again.

File: include/types.h [hunk 12]
```diff
@@ -344,6 +344,7 @@ struct MarioState
     /*0xBC*/ f32 peakHeight;
     /*0xC0*/ f32 quicksandDepth;
     /*0xC4*/ f32 unkC4;
+             u8 pretiptimer;
```
Comment (Fluvian) [inline include/types.h:347]: Change this to `pretipTimer` as mentioned in the other comment.

File: TPPCHANGELIST.txt [hunk 13]
```diff
@@ -0,0 +1,45 @@
+TPP CHANGES LIST:
```
Comment (Fluvian) [inline TPPCHANGELIST.txt:1]: This file can be removed as it doesn't need to be in the repository, I will use it as a commit description.

File: levels/bowser_1/areas/1/1/model.inc.c [hunk 14]
```diff
@@ -1,210 +1,1263 @@
-// 0x07002000 - 0x07002018
-static const Lights1 bowser_1_seg7_lights_07002000 = gdSPDefLights1(
-    0x3f, 0x3f, 0x3f,
-    0xff, 0xff, 0xff, 0x28, 0x28, 0x28
-);
+Lights1 bowser_1_f3d_material_004_lights = gdSPDefLights1(
```
Comment (Fluvian) [inline levels/bowser_1/areas/1/1/model.inc.c:1]: Why is the whole level re-exported with Fast64 just to remove the fog? Just revert this, I will remove the fog myself after this pull request gets merged.

**Review by Fluvian — CHANGES_REQUESTED** [review]

**Comment by Fluvian** [comment]
If these changes are made and I make the changes I mentioned I'll release it as the second release.

File: src/game/mario_step.c [hunk 15]
```diff
@@ -532,7 +532,7 @@ void apply_gravity(struct MarioState *m) {
     if (m->action == ACT_TWIRLING && m->vel[1] < 0.0f) {
         apply_twirl_gravity(m);
     } else if (m->action == ACT_SHOT_FROM_CANNON) {
-        m->vel[1] -= 3.2f;
+        m->vel[1] -= 0.0f;
```
[the remark this answers is not in the document]
Comment (TorutheRedFox) [reply src/game/mario_step.c:535]: yep, if you're subtracting nothing, why bother have an op for it

File: src/game/mario_actions_cutscene.c [hunk 16]
```diff
@@ -1051,7 +1051,7 @@ s32 act_exit_land_save_dialog(struct MarioState *m) {
 
 s32 act_death_exit(struct MarioState *m) {
     if (m->actionTimer == 15) {
```
Comment (TorutheRedFox) [inline src/game/mario_actions_cutscene.c:1053]: having brackets here isn't needed as it's a single line if statement

**Comment by The1UPGuy** [comment]
Oh wait there's a few more lol, GitHub didn't show them to me

**Comment by The1UPGuy** [comment]
I believe that's everything. Thanks for the feedback!

**Pull request merged**